server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSStartbrowseOptionsCheckUtility.java

```diff
@@ -0,0 +1,114 @@
+        {
+            put(CICSLexer.ACTIVITYID, ErrorSeverity.ERROR);
+            put(CICSLexer.PROCESS, ErrorSeverity.ERROR);
+            put(CICSLexer.PROCESSTYPE, ErrorSeverity.ERROR);
+            put(CICSLexer.ACTIVITY, ErrorSeverity.WARNING);
+            put(CICSLexer.BROWSETOKEN, ErrorSeverity.ERROR);
+            put(CICSLexer.CONTAINER, ErrorSeverity.WARNING);
+            put(CICSLexer.CHANNEL, ErrorSeverity.WARNING);
+            put(CICSLexer.EVENT, ErrorSeverity.WARNING);
+            put(CICSLexer.TIMER, ErrorSeverity.WARNING);
+            put(CICSLexer.RETCODE, ErrorSeverity.ERROR);
+        }
+    };
+
+    /**
+     * Checks CICS Startbrowse rules for required and invalid options
+     * @param dialectProcessingContext
+     * @param errors
+     */
+    public CICSStartbrowseOptionsCheckUtility(
+        DialectProcessingContext dialectProcessingContext,
+        List<SyntaxError> errors) {
+      super(dialectProcessingContext, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS STARTBROWSE rule options
+     *
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        switch (ctx.getRuleIndex()) {
+            case CICSParser.RULE_cics_startbrowse_activity:
+                checkActivity((CICSParser.Cics_startbrowse_activityContext) ctx);
+                break;
+            case CICSParser.RULE_cics_startbrowse_container:
+                checkContainer((CICSParser.Cics_startbrowse_containerContext) ctx);
+                break;
+            case CICSParser.RULE_cics_startbrowse_event:
+                checkEvent((CICSParser.Cics_startbrowse_eventContext) ctx);
+                break;
+            case CICSParser.RULE_cics_startbrowse_process:
+                checkProcess((CICSParser.Cics_startbrowse_processContext) ctx);
+                break;
+            case CICSParser.RULE_cics_startbrowse_timer:
+                checkTimer((CICSParser.Cics_startbrowse_timerContext) ctx);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void checkActivity(CICSParser.Cics_startbrowse_activityContext ctx) {
+        checkHasMandatoryOptions(ctx.BROWSETOKEN(), ctx, "BROWSETOKEN");
+        checkPrerequisiteIsMet(ctx.PROCESS(), ctx.PROCESSTYPE(), ctx, "PROCESSTYPE without PROCESS");
+        checkMutuallyExclusiveOptions("ACTIVITYID or PROCESS", ctx.ACTIVITYID(), ctx.PROCESS());
+    }
+
+    private void checkContainer(CICSParser.Cics_startbrowse_containerContext ctx) {
+        checkHasMandatoryOptions(ctx.BROWSETOKEN(), ctx, "BROWSETOKEN");
+        checkPrerequisiteIsMet(ctx.PROCESS(), ctx.PROCESSTYPE(), ctx, "PROCESSTYPE without PROCESS");
+        checkMutuallyExclusiveOptions("ACTIVITYID, PROCESS or CHANNEL", ctx.ACTIVITYID(), ctx.PROCESS(), ctx.CHANNEL());
+    }
```

Comment:
`CHANNEL` requires `CONTAINER`.

Also the more I look at this the more I am wondering if keeping the approach with 5 grammar rules for the options is a good idea given the similarity between them.